mmv1/third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -1713,6 +1713,23 @@ func resourceContainerCluster() *schema.Resource {
 					},
 				},
 			},
+			"gateway_api_config": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MaxItems:    1,
+				Description: `Configuration for GKE Gateway API controller.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"channel": {
+							Type:         schema.TypeString,
+							Default:      "CHANNEL_UNSPECIFIED",
+							ValidateFunc: validation.StringInSlice([]string{"CHANNEL_UNSPECIFIED", "CHANNEL_DISABLED", "CHANNEL_EXPERIMENTAL", "CHANNEL_STANDARD"}, false),
+							Description:  `The Gateway API release channel to use for Gateway API.`,
+							Optional:     true,
```

Comment:
Just to double-check, is this field actually optional? I.e. if I hypothetically sent an empty value for `gateway_api_config` would that be a valid / meaningful request?

It might be reasonable to say that this field is required if the parent block is defined. That would also mean you wouldn't need to set `Default`

```suggestion
```

Reply:
I decided to make it a required field as setting it to `CHANNEL_UNSPECIFIED` by default or leaving it all together will just lead to the same effect as not adding the `gateway_api_config` block at all

Reply:
@melinath 
I checked all channels by hand, with the GA and Beta provider versions which led me to the following decision:

Despite being mentioned in the comments of the Go library the experimental channel does not actually exist and will just throw `Error: googleapi: Error 400: Unsupported Gateway API release channel., badRequest`. Therefore I removed it, furthermore as the unspecified channel does not really have a purpose and is virtual the same as just removing the `gateway_api_config` block completely or setting the channel to `CHANNEL_DISABLED` I removed it as well.